backend/kesaseteli/applications/tests/test_schools_api.py

```diff
@@ -0,0 +1,61 @@
+import pytest
+from rest_framework import status
+
+
+def get_schools_api_url():
+    return "/v1/schools/"
+
+
+@pytest.mark.django_db
+def test_schools_list_status_ok(api_client):
+    response = api_client.get(get_schools_api_url())
+    assert response.status_code == status.HTTP_200_OK
+
+
+@pytest.mark.django_db
+def test_schools_list_returns_list(api_client):
+    response = api_client.get(get_schools_api_url())
+    assert type(response.json()) == list
+
+
+@pytest.mark.django_db
+def test_schools_list_returns_non_empty_list(api_client):
+    response = api_client.get(get_schools_api_url())
+    assert type(response.json()) == list
+    assert len(response.json()) > 0
+
+
+@pytest.mark.django_db
+def test_schools_list_returns_non_empty_string_list(api_client):
+    response = api_client.get(get_schools_api_url())
+    assert type(response.json()) == list
+    assert len(response.json()) > 0
+    assert all(type(element) == str for element in response.json())
+
+
+@pytest.mark.django_db
+def test_schools_list_returns_non_empty_list_of_non_empty_strings(api_client):
+    response = api_client.get(get_schools_api_url())
+    assert response.status_code == status.HTTP_200_OK
+    assert type(response.json()) == list
+    assert len(response.json()) > 0
+    assert all(type(element) == str for element in response.json())
+    assert all(len(element.strip()) > 0 for element in response.json())
+
+
+@pytest.mark.django_db
+def test_schools_list_returns_non_empty_list_of_unique_elements(api_client):
+    response = api_client.get(get_schools_api_url())
+    assert response.status_code == status.HTTP_200_OK
+    assert type(response.json()) == list
+    assert len(response.json()) > 0
+    assert len(set(response.json())) == len(response.json())
+
+
+@pytest.mark.django_db
+def test_schools_list_returns_non_empty_sorted_list(api_client):
```

Comment:
General comment about these tests. They look quite duplicated to me, seems like the next test is a superset of the previous test. For example, while nothing changes in the request, you don't have to assert `assert type(response.json()) == list` in the second test if the first test already did that. Each test should be isolated and test a specific expectation

Reply:
I'll make the tets more isolated.